sky/templates/kubernetes-ray.yml.j2

```diff
@@ -322,18 +313,91 @@ available_node_types:
         - name: ray-node
           imagePullPolicy: IfNotPresent
           image: {{image_id}}
+          env:
+            - name: SKYPILOT_POD_NODE_TYPE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.labels['ray-node-type']
           # Do not change this command - it keeps the pod alive until it is
           # explicitly killed.
+          # We need to use -i to make sure the environment variables are set.
           command: ["/bin/bash", "-c", "--"]
           args: 
             - |
               # Helper function to conditionally use sudo
               prefix_cmd() { if [ $(id -u) -ne 0 ]; then echo "sudo"; else echo ""; fi; }
+              
+              (
+                {
+                mkdir -p ~/.sky
+                {{ conda_installation_commands }}
+                {{ ray_installation_commands }}
+                ~/skypilot-runtime/bin/python -m pip install skypilot[kubernetes,remote]
+                touch /tmp/ray_skypilot_installation_complete
+                echo "=== Ray and skypilot installation completed ==="
+                
+                if [ "$SKYPILOT_POD_NODE_TYPE" == "head" ]; then
+                  {{ ray_head_start_command }}
+                  touch /tmp/ray_started
+                else
+                  # Start ray worker on the worker pod.
+                  # Wait until the head pod is available with an IP address
+                  export SKYPILOT_RAY_HEAD_IP="{{cluster_name_on_cloud}}-head.{{k8s_namespace}}.svc.cluster.local"
+                  export SKYPILOT_RAY_PORT={{skypilot_ray_port}}
+                  # Wait until the ray cluster is started on the head pod
+                  until nc -z -w 1 ${SKYPILOT_RAY_HEAD_IP} ${SKYPILOT_RAY_PORT}; do
+                    sleep 0.1
+                  done
+
+                  {{ ray_worker_start_command }}
+                  touch /tmp/ray_started
+                fi
+                } | tee /tmp/runtime-setup.log 2>&1
```

Comment:
I think we should keep the container's log free of SkyPilot setup commands (since users may use it for getting task outputs). wdyt about redirecting all output to file and printing to stdout only if there's an error?

Currently:
```
$ kubectl logs -f test-2ea4-head
++ id -u
/bin/bash: line 5: curl: command not found
+ '[' 0 -eq 0 ']'
+ echo 'alias sudo=""'
+ echo succeed
succeed
/bin/bash: line 5: conda: command not found
+ printenv
+ IFS==
+ read -r key value
+ echo 'export HOSTNAME="test-2ea4-head"'
+ IFS==
+ read -r key value
/bin/bash: line 5: python3: command not found
+ echo 'export TEST_2EA4_HEAD_SSH_PORT_22_TCP_PORT="22"'
+ IFS==
+ read -r key value
+ echo 'export TEST_2EA4_HEAD_SSH_SERVICE_HOST="34.118.238.236"'
+ IFS==
+ read -r key value
+ echo 'export TEST_2EA4_HEAD_SSH_PORT_22_TCP="tcp://34.118.238.236:22"'
+ IFS==
+ read -r key value
+ echo 'export TEST_2EA4_HEAD_SSH_SERVICE_PORT="22"'
+ IFS==
+ read -r key value
+ echo 'export KUBERNETES_PORT_443_TCP_PROTO="tcp"'
+ IFS==
+ read -r key value
+ echo 'export KUBERNETES_PORT_443_TCP_ADDR="34.118.224.1"'
+ IFS==
+ read -r key value
+ echo 'export KUBERNETES_PORT="tcp://34.118.224.1:443"'
+ IFS==
+ read -r key value
+ echo 'export PWD="/"'
+ IFS==
+ read -r key value
+ echo 'export HOME="/root"'
+ IFS==
+ read -r key value
+ echo 'export KUBERNETES_SERVICE_PORT_HTTPS="443"'
+ IFS==
+ read -r key value
+ echo 'export KUBERNETES_PORT_443_TCP_PORT="443"'
+ IFS==
+ read -r key value
+ echo 'export TEST_2EA4_HEAD_SSH_PORT_22_TCP_PROTO="tcp"'
+ IFS==
+ read -r key value
+ echo 'export SKYPILOT_POD_NODE_TYPE="head"'
+ IFS==
+ read -r key value
+ echo 'export KUBERNETES_PORT_443_TCP="tcp://34.118.224.1:443"'
+ IFS==
+ read -r key value
+ echo 'export TEST_2EA4_HEAD_SSH_PORT="tcp://34.118.238.236:22"'
+ IFS==
+ read -r key value
+ echo 'export SHLVL="1"'
+ IFS==
+ read -r key value
+ echo 'export KUBERNETES_SERVICE_PORT="443"'
+ IFS==
+ read -r key value
+ echo 'export PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"'
+ IFS==
+ read -r key value
+ echo 'export TEST_2EA4_HEAD_SSH_PORT_22_TCP_ADDR="34.118.238.236"'
+ IFS==
+ read -r key value
+ echo 'export KUBERNETES_SERVICE_HOST="34.118.224.1"'
+ IFS==
+ read -r key value
+ echo 'export _="/usr/bin/printenv"'
+ IFS==
+ read -r key value
++ prefix_cmd
+++ id -u
/bin/bash: line 5: -m: command not found
++ '[' 0 -ne 0 ']'
++ echo ''
+ mv /root/container_env_var.sh /etc/profile.d/container_env_var.sh
Nproc: 192
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
/bin/bash: line 8: /root/skypilot-runtime/bin/python: No such file or directory
/bin/bash: line 8: /root/skypilot-runtime/bin/python: No such file or directory
/bin/bash: line 8: /root/skypilot-runtime/bin/python: No such file or directory
/bin/bash: line 8: /root/skypilot-runtime/bin/python: No such file or directory
/bin/bash: line 8: /root/skypilot-runtime/bin/activate: No such file or directory
== Installed ray: 0 secs ==
Installing missing packages:  gcc patch pciutils rsync fuse curl openssh-server rsync
/bin/bash: line 47: sudo: command not found
sed: can't read /etc/ssh/sshd_config: No such file or directory
sed: can't read /etc/pam.d/sshd: No such file or directory
/bin/bash: line 52: cd: /etc/ssh/: No such file or directory
ssh: unrecognized service
```

Reply:
Good point! fixed.